Updates to ParmEst documentation (#3791)
Pyomo/pyomo: doc/OnlineDocs/explanation/analysis/parmest/driver.rst

   * - hour
     - y
   * - 1
     - 8.3
   * - 2
     - 10.3
   * - 3
     - 19.0
   * - 4
     - 16.0
   * - 5
     - 15.6
   * - 7
     - 19.8

To use parmest to estimate :math:`\theta_1` and :math:`\theta_2` from the data, we provide the following
detailed steps:

Step 0: Import Pyomo, parmest, Experiment Class, and Pandas
-----------------------------------------------------------

Before solving the parameter estimation problem, the following code must be executed to import the
required packages for parameter estimation in parmest:

.. doctest::
    :skipif: not __import__('pyomo.contrib.parmest.parmest').contrib.parmest.parmest.parmest_available

    >>> import pyomo.environ as pyo
    >>> import pyomo.contrib.parmest.parmest as parmest
    >>> from pyomo.contrib.parmest.experiment import Experiment
    >>> import pandas as pd

.. _ExperimentClass:

Step 1: Create the Experiment Class for the Model
-------------------------------------------------

parmest requires that the user creates an :class:`~pyomo.contrib.parmest.experiment.Experiment` class that
contains the annotated Pyomo model with labeled experiment outputs, unknown parameters, and measurement errors.

> A labeled Pyomo model ``m`` has the following additional suffixes (Pyomo `Suffix`):

Reviewer: ```suggestion
parmest requires that the user create an :class:`~pyomo.contrib.parmest.experiment.Experiment` class that
builds an annotated Pyomo model denoting experiment outputs, unknown parameters, and measurement errors using Pyomo `Suffix` components.
```

* ``m.experiment_outputs`` which defines experiment output (Pyomo `Param`, `Var`, or `Expression`)
  and their associated data values (float, int).
* ``m.unknown_parameters`` which defines the mutable parameters or variables (Pyomo `Param` or `Var`)

>   to estimate along with their component unique identifier (Pyomo `ComponentUID`).

Reviewer: You should include `m.measurement_error` in this list

Author: Thanks for catching this.

>   to estimate along with their component unique identifier (Pyomo `ComponentUID`).

Reviewer: Rewording suggestion:
```suggestion
* ``m.experiment_outputs`` maps the experiment output, or measurement, terms in the model (Pyomo `Param`, `Var`, or `Expression`) to their associated data values (float, int).
* ``m.unknown_parameters`` maps the model parameters to estimate (Pyomo `Param` or `Var`)
  to their component unique identifier (Pyomo `ComponentUID`) which is used to identify equivalent parameters across multiple experiments.
```

    ...
    ...         return m
    ...
    ...     def label_model(self):
    ...
    ...         m = self.model
    ...
    ...         # label the experiment outputs
    ...         m.experiment_outputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
    ...         m.experiment_outputs.update([(m.y[self.data['hour']], self.data['y'])])
    ...
    ...         # label the unknown parameters in the model
    ...         m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
    ...         m.unknown_parameters.update(
    ...             (k, pyo.ComponentUID(k)) for k in [m.asymptote, m.rate_constant]
    ...         )
    ...
    ...         # add the measurement error assumed to be constant and 0.1
    ...         m.measurement_error = pyo.Suffix(direction=pyo.Suffix.LOCAL)
    ...         m.measurement_error.update([(m.y[self.data['hour']], 0.1)])
    ...
    ...     def finalize_model(self):
    ...
    ...         m = self.model
    ...         pass
    ...
    ...     def get_labeled_model(self):
    ...         self.create_model()
    ...         self.label_model()
    ...         self.finalize_model()
    ...

>     ...         return self.model

Reviewer: Please replace this with a literal include block that pulls the experiment class directly from the py file in the repo. Duplicating the code here makes it too easy for things to become out of sync.

Author: Got it. Thanks.

> Step 2: Load the Data and Create a List of the Model's Experiment Class

Reviewer: Suggested rewording:
```suggestion
Step 2: Load the Data and Create a List Experiments
```

This step shows how to create the :class:`~pyomo.contrib.parmest.experiment.Experiment` class using the
mathematical model outlined in the introduction section of this Quick Start.

> .. literalinclude:: ../../../pyomo/contrib/parmest/examples/rooney_biegler/rooney_biegler.py

Reviewer: This isn't rendering in the documentation. I think the path is slightly off. 
```suggestion
.. literalinclude:: ../../pyomo/contrib/parmest/examples/rooney_biegler/rooney_biegler.py
```

    >>> solver_options = {"max_iter": 6000}
    >>> pest = parmest.Estimator(exp_list, obj_function="SSE", solver_options=solver_options)

Objective function
^^^^^^^^^^^^^^^^^^

The second argument is an optional argument if the objective function has already been included in the
Pyomo model, which defines the optimization objective function to use in parameter estimation. However, if the
objective function has not been included in the Pyomo model, like the one in the :ref:`ExperimentClass` Section

> above, the user is required to supply the second argument.

Reviewer: Suggested rewording:
```suggestion
The second argument, ``obj_function``, is used to specify the objective function to use for parameter estimation if the user has not manually defined an objective function in their ``Experiment`` class.
```

> If no objective function is specified, the Pyomo model is used "as is" and

Reviewer: ```suggestion
If ``obj_function`` is not specified, the Pyomo model is used "as is" and
```
